OpenSourcePolitics/decidim-app
# Fix custom sort for processes missing date

#### :tophat: Description
*Please describe your pull request.*

#### :pushpin: Related Issues
- Related to #https://github.com/OpenSourcePolitics/decidim-app/issues/619
- Related to #https://github.com/OpenSourcePolitics/decidim-app/pull/596
- [Odoo card](https://opensourcepolitics.odoo.com/web#id=3374&cids=1&menu_id=325&action=471&model=project.task&view_type=form)

#### Testing

1. As an admin, create a participatory process with start date before today and without end date, and one without start date and end date, and publish them
2. As an admin, go to an assembly, add all Related participatory processes available and update.
3. As a user, go to the show page of the assembly and see that Related participatory processes are not sorted 
4. Stop your server and relaunch it with SORT_PROCESSES_BY_DATE=true bundle exec rails s
5. As a user, go to the show page of the assembly and see that Related participatory processes are sorted with:
 - active first (sorted by end date, with processes without end date last)
 - future second (sorted by start date)
 - past third (sorted by end date reverse)
6. As a user, go to the processes, check that active processes are sorted by end date, with processes without end date last
7. Change for upcoming and see that processes are sorted by start date
8. Change for past and see that processes are sorted by end date reverse

#### Tasks
- [X ] Add specs
